### ISSUE TITLE
Searching a new maintainer

### ISSUE THREAD

**Issue by johanjanssen** [johanjanssen/TemurinChocolateyPackages#26]

**Comment by johanjanssen** [comment]
@RedBaron2 you helped me a lot with the Temurin packages. In case you're interested in taking over the maintenance, please let me know.

**Comment by RedBaron2** [comment]
@johanjanssen 
I have no problem working on the code for fixes, but being the person in charge of pushing the package. I'll need to think about it.

**Comment by johanjanssen** [comment]
Yes of course and don't feel forced to do it, no is also a perfectly fine answer :).

**Comment by untra** [comment]
I would not be opposed to stepping up and taking ownership of the openjdk chocolatey packages. I saw you just updated the package to temurin11.

I don't develop on windows normally, but I work for a company that would like to see this software properly maintained and this is a significant piece of OSS the windows community should be able to rely on.

**Comment by johanjanssen** [comment]
The OpenJDK Chocolatey packages used the upstream builds from AdoptOpenJDK. However those are no longer created, for more info see: https://adoptopenjdk.net/upstream.html. Now the OpenJDK Chocolatey packages simply install Temurin. I've just configured this yesterday, which basically is a one time change. The OpenJDK packages, are now deprecated. I recommend everyone using them to explicitly move to another JDK such as Temurin.

So thanks for the offer, but there's not really any work left for the Chocolatey OpenJDK packages.

**Comment by ajshastri** [comment]
@johanjanssen If you are still looking for maintainers of the Temurin packages, I can do that. I've adopted the Zulu java ones, so I assume it's not very much different.

**Comment by johanjanssen** [comment]
I performed the update for the new major version and the minor versions and described the process here: https://github.com/johanjanssen/TemurinChocolateyPackages.

Basically if you copy my GitHub actions it works out of the box, only for new major versions you need to change the script.

Shall I make you co-maintainer? Then you can copy the configuration to your repository, change the configuration and then next time release from your repository.

BTW I have some more Java Chocolatey packages https://github.com/johanjanssen/ChocolateyPackages, if you're interested in maintaining them as well, please let me know. They don't use au to update the package, some can be updated with a Java applications, others are currently manual updated.

**Comment by ajshastri** [comment]
@johanjanssen I've reviewed the README; thank you for making it super easy to understand. I think I'll copy the config over to my repo as is, just as you suggested, and I'll release it from my repo next time. A co-maintainer would be perfect.

As for the remaining Java Chocolatey packages, I can take them over as well. I've finally got my Zulu Nuspec to update when a new version is available and in the next release have it auto-deploy to Chocolatey. I'll try to make sure they all do so there's minimal work in the long term.

**Comment by johanjanssen** [comment]
Great, just to double check, linearreg is your username at Chocolatey?

I'll make you co-maintainer of all packages, as soon as you're fully setup I will remove myself as maintainer :).

I got quite some packages, I'll add you for all the major Java versions, the others are anyhow no longer maintained and the Microsoft ones will be picked up by someone else. Please have a look at https://community.chocolatey.org/profiles/JohanJanssen if there are any other packages you don't want, then I'll try to find someone else for them.

**Comment by ajshastri** [comment]
linearreg is me, yes. That sounds good to me on the maintainer plan.

I took a look at the packages that you have and I think I can handle the various Java ones.

I'll take the next two months to fully integrate them into my repo in time for the next big security patch. :)

**Comment by johanjanssen** [comment]
Just made you a maintainer, if you think you miss one, please let me know. Also if you have any questions don't hesitate to contact me!

I can also advice you to have a look at the GitHub actions for https://github.com/johanjanssen/ChocolateyPackages. They are different then the ones for Temurin and can be used with the Java app https://github.com/johanjanssen/ChocolateyPackages/tree/master/updater to update some of the packages in that repository.

Maybe https://github.com/foojayio/discoapi is an improvement you can consider, it provides API's for all vendors. My solution uses the GitHub API's, but not all vendors support that.

**Comment by ajshastri** [comment]
Thank you, I'll work on this over the next month and have it done.

https://github.com/foojayio/discoapi Looks like an excellent solution! I will work to see if we can incorporate it into a unified source. Github's API was going to be my go-to as well but this may be simpler.

**Comment by ajshastri** [comment]
@johanjanssen cloned all the chocolatey packages you had and have written automation scripts to keep them updated for most except for Graalvm and one off stuff not available from foojay. I'll work to fix that too.

I have a quick question about Graalvm, Graalvm 17 seems to point to the "vm-22.3.2" (vm-22.3.3 appears to be the last one which was released in Jun 2023 )  release while the v21 points to their JDK.  The latest of the "vm" appears to not have any assets which are not zip files. Do you think it's a good idea to move v17 to it's JDK as well?

**Comment by johanjanssen** [comment]
@ajshastri great that you managed to automate them!

Regarding GraalVM, I mostly look at https://github.com/graalvm/graalvm-ce-builds/releases/ or https://github.com/graalvm/graalvm-ce-builds/tags for the latest version. It looks like they stopped working with 'vm' tags and they're now using the JDK versions instead of their own versioning.

Currently the latest Chocolatey for Java 17 https://community.chocolatey.org/packages/graalvm-java17 has version 22.3.2. If you move to the latest version of GraalVM https://github.com/graalvm/graalvm-ce-builds/releases/tag/jdk-17.0.9 it has version 17.0.9. It's lower then the current version, so that won't work, you can't upload it and you can't remove existing versions. I guess the easiest 'work around' is to create new Chocolatey packages with another id, for example graalvm-community-java17.

**Comment by johanjanssen** [comment]
@ajshastri seems like everything is working with the Chocolatey packages? So I can remove myself as a maintainer?

**Comment by ajshastri** [comment]
For sure, thanks for your help!

**Comment by johanjanssen** [comment]
@ajshastri looks like I can't remove myself from the packages. So whenever you have time, please remove me from the packages :). Thanks a lot for taking over the maintenance of the packages!

**Comment by ajshastri** [comment]
@johanjanssen please confirm that you have been removed from all packages assigned to me.

**Comment by johanjanssen** [comment]
Yes looks like it's only old stuff which is left. Again thanks for taking over the packages @ajshastri and have fun maintaining them :).
